libcrux-ml-kem/src/vector/portable/arithmetic.rs

```diff
@@ -422,3 +422,29 @@ pub(crate) fn montgomery_multiply_by_constant(mut vec: PortableVector, c: i16) -
     }
     vec
 }
+
+#[hax_lib::requires(fstar!(r#"Spec.Utils.is_i16b_array 3328 ${a}.f_elements"#))]
+#[hax_lib::ensures(|result| fstar!(r#"forall i.
+                                       (let x = Seq.index ${a}.f_elements i in
+                                        let y = Seq.index ${result}.f_elements i in
+                                        (v y >= 0 /\ v y <= 3328 /\ (v y % 3329 == v x % 3329)))"#))]
+#[inline(always)]
+pub(crate) fn to_unsigned_representative(a: PortableVector) -> PortableVector {
+    let t = shift_right::<15>(a);
+    hax_lib::fstar!(
+        r#"
+  assert (forall i. Seq.index ${t}.f_elements i == ((Seq.index ${a}.f_elements i) >>! (mk_i32 15)));
```

Comment:
Please separate the proof code from the actual code with empty lines.

Reply:
Doing a best-effort separation, but we should try to find a pleasant style and implement it in every file we touch.